foojay-resolver/src/main/kotlin/org/gradle/toolchains/foojay/distributions.kt

```diff
@@ -30,24 +30,34 @@ fun match(
         vendor: JvmVendorSpec,
         implementation: JvmImplementation,
         version: JavaLanguageVersion
-): Distribution? {
-    if (JvmImplementation.J9 == implementation) {
-        return distributions.firstOrNull { it.name == j9Aliases[vendor] }
-    }
-    if (JvmVendorSpec.GRAAL_VM == vendor) {
-        return match(distributions, JvmVendorSpec.matching("Graal VM CE " + version.asInt()))
+): List<Distribution> = when {
+    JvmImplementation.J9 == implementation -> distributions.filter { it.name == j9Aliases[vendor] }
+    JvmVendorSpec.GRAAL_VM == vendor -> match(distributions, JvmVendorSpec.matching("Graal VM CE " + version.asInt()))
+    else -> match(distributions, vendor)
+}
+
+fun match(distributions: List<Distribution>, vendor: JvmVendorSpec): List<Distribution> {
+    if (vendor == any()) {
```

Comment:
Return everything, but put Temurin first, and Graal VM last. Graal VM needs to be filtered, because it's Java language level specific.